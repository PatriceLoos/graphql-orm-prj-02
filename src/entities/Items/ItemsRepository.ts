import { EntityRepository, Repository } from 'typeorm';
import Items from '.';
import CreateItemParams from '../../graphql/resolvers/Items/types/inputs/CreateItem';

@EntityRepository(Items)
export default class ItemsRepository extends Repository<Items> {
  async getItems() {
    return this.find();
  }

  async createItem(params: CreateItemParams):Promise<any> {
    return this.save(params).catch((err)=>{
      console.log(`Error creating item ${
        Object.keys(params)}:${
          Object.values(params)}, with error ${err}`)
    });
  }

  async updateItem(id:string,params: CreateItemParams):Promise<any> {
    return this.update(id,params).catch((err)=>{
      console.log(`Error updating item ${
        Object.keys(params)}:${
          Object.values(params)}, with error ${err}`)
    });
  }

  async deleteItem(param: {}):Promise<any> {
    return this.delete(param).catch((err)=>{
      console.log(`Error deleting item${
        Object.keys(param)}:${
          Object.values(param)}, with error ${err}`)
    });
  }

}
