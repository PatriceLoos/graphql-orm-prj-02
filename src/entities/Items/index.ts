import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity({ name: 'items' })
export default class Items {

  @PrimaryGeneratedColumn()
  id!: number;

   @Column({ type: 'text', nullable: true })
  name?: string;

  @Column({ type: 'text', nullable: true })
  address?: string;

   @Column({ type: 'text', nullable: true })
  description?: string;  

   @Column({ type: 'text', nullable: true })
  brand?: string;

   @Column({ type: 'text', nullable: true })
  make?: string;

   @Column({ type: 'text', nullable: true })
  model?: string;

   @Column({ type: 'text', nullable: true })
  size?: string;

   @Column({ type: 'text', nullable: true })
  genderUsage?: string;

   @Column({ type: 'text', nullable: true })
  type?: string;

   @Column({ type: 'text', nullable: true })
  identifier?: string;

  @Column({ type: 'text', nullable: true })
  ownerEmail?: string;

  @Column({ type: 'date', name: 'stolen_date' })
  dod?: string;

  @Column({ type: 'date', name: 'production_date' , nullable: true })
  dop?: string;
  
  @CreateDateColumn({
    type: 'timestamptz',
    name: 'created_at',
    default: () => 'CURRENT_TIMESTAMP',
  })
  createdAt!: Date;

  @CreateDateColumn({
    type: 'timestamptz',
    name: 'updated_at',
    default: () => 'CURRENT_TIMESTAMP',
  })
  updatedAt!: Date;

   @Column({ type: 'text', nullable: true })
  imageUrl?: string;

}
