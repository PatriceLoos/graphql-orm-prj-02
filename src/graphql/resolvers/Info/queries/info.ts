import { Resolver, Query } from 'type-graphql';
import AppInfo from '../types/objects/Info';
import { environment } from '../../../../environment';

@Resolver()
class InfoResolver {
  @Query(() => AppInfo, { description: 'API info' })
  info(): AppInfo {
    return {
      port: Number(environment.PORT),
      healthy: true,
    };
  }
}

export default InfoResolver;
