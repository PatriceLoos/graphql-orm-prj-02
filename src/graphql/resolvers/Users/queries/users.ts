import { Resolver, Query } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Users from '../types/objects/Users';
import UserRepository from '../../../../entities/Users/UserRepository';

@Resolver()
export default class GetUserResolver {
  @InjectRepository(UserRepository)
  private UserRepository!: UserRepository;

  @Query(() => [Users], { description: 'List of Users' })
  async users(): Promise<Users[]> {
    return this.UserRepository.getUser();
  }
}
