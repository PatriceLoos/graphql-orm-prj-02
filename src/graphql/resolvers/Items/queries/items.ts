import ItemsRepository from '../../../../entities/Items/ItemsRepository';
import { Resolver, Query } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Items from '../types/objects/Items';


@Resolver()
export default class GetItemsResolver {
  @InjectRepository(ItemsRepository)
  private itemsRepository!: ItemsRepository;

  @Query(() => [Items], { description: 'List of items' })
  async items(): Promise<Items[]> {
    return this.itemsRepository.getItems();
  }
}
