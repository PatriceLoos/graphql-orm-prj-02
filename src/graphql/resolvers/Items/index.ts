import items from './queries/items';
import createItem from './mutations/items';

export default {
  items,
  createItem,
};
