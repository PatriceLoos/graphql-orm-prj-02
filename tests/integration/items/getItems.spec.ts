import { Connection } from 'typeorm';
import { createTestClient } from 'apollo-server-testing';
import { getItemsQuery } from './__gql__/queries';
import {
  createTestConnection,
  createTestServer,
  clearTestDb,
} from '../../utils';
import Items from '../../../src/entities/Items';

let testConnection: Connection;

beforeAll(async () => {
  testConnection = await createTestConnection();
});

beforeEach(async () => {
  await clearTestDb(testConnection);
});

afterAll(async () => {
  await testConnection.close();
});

describe('Get Items', () => {
  it('should return empty array if no items are registered', async () => {
    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getItemsQuery,
    });
    expect(res).toMatchSnapshot('Empty items list');
  });

  it('should return array with one registered item', async () => {
    const item = {
      id: 1,
      name: "cravache",
      address: "2020 ouille st, St-Louis, MS, 020102",
      description: "whip",
      brand: "Fischer",
      make: "Aerospace",
      model: "brown suede",
      size: "unisize",
      genderUsage: "n/a",
      type: "",
      identifier: "8787878",
      ownerEmail: "d@zone.com",
      dod: "08/29/2010",
      dop: "12/12/2001",
      imageUrl: "www.url.gov",
    };

    await testConnection.getRepository(Items).save(item);

    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getItemsQuery,
    });
    expect(res).toMatchSnapshot('Filled items list');
  });
});
