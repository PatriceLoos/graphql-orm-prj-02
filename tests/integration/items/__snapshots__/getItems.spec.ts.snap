// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Items should return array with one registered item: Filled items list 1`] = `
Object {
  "data": Object {
    "items": Array [
      Object {
        "address": "2020 ouille st, St-Louis, MS, 020102",
        "brand": "Fischer",
        "description": "whip",
        "dod": "2010-08-29",
        "dop": "2001-12-12",
        "genderUsage": "n/a",
        "id": 1,
        "identifier": "8787878",
        "imageUrl": "www.url.gov",
        "make": "Aerospace",
        "model": "brown suede",
        "name": "cravache",
        "ownerEmail": "d@zone.com",
        "size": "unisize",
        "type": "",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Items should return empty array if no items are registered: Empty items list 1`] = `
Object {
  "data": Object {
    "items": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
